unit-tests/src/test/scala/scala/scalanative/posix/TimeTest.scala

```diff
@@ -20,14 +21,26 @@ class TimeTest {
   val now_time_t: time_t = scala.scalanative.posix.time.time(null)
   val epoch: time_t      = 0L
 
+  // some of the tests (the ones that call localtime) need
```

Comment:
Ah,  the pain point.   We, or our descendents, are almost certainly going to 
be back here again. even after having given this evolution our best shot.
I think it is worth leaving some clues behind (beyond weathered, broken bones).

I suggest something like the text below, suitably cleaned up.
Yes, this is a __long__ comment, but think of all the time
spent on this topic already.

The alternative is to put a 2 line comment in the PR file, pointing
to the longer discussion below.   IMO, not as useful to the next
poor devil who has to debug this on the fly, but more likely to 
pass Scala Center review on the first pass.

=========================================================


The behavior of the called C RTL with respect to the handling of the
`tm.isdst` field can and does vary across implementations. It is hard
to discover and enumerate the cases. 

Some C RTLs, such as Ubuntu
glibc, appear to use the daylight saving value at the time of the supplied
`time_t`.  Others appear to use the daylight saving value of the currently
executing program.  The Posix standard provides little, if any, guidance.

The purpose of these tests is to show that the Scala Native code
is calling the correct entry points, arguments correctly, and getting
reasonable return values. It is not to prove out fine implementation 
details of every C RTL implementation.

The Continuous Integration (CI) environment operates in a known, constant
UTC timezone.  So timeIsStandard will always return Boolean true and
the tests will execute.  That gives the confidence that the tests
executed correctly on at least one system.

Executing this Test on non-CI systems expands the circle of confidence
but only where it can be done reliably.  Executing the localtime*
tests "in the wild" only when timeIsStandard is false has been shown
to be unreliable when considered over a large, varied population of
C RTL implementations.

Reply:
@ncordon Thank you for your time & persistence on this issue.  I think we
are converging on a solution.

I like the current concept and handling of the `localtime*()` tests. 

Please consider the review comments I left.  My intent was to 
help another evolution.  I think deleting the two extraneous (and 
probably un-intentional) files is the minimal change.

Reply:
I've flipped the sense of the assumptions as requested, but I believe the comment is not accurate. It does not depend on the underlying stdlib as long as I've tested, just on one's own timezone.

Reply:
If you/we have not exhausted patience, I think it is worth seeing if we can nail the
varying condition(s).    If that fails, we can always punt and say "Under unknown and
difficult to find conditions in the wild".  Comes a time, sometimes, when the best that
can be is to mark something "Done" and hope that the passage of time
brings increased understanding.  Due diligence in effort , not exhaustion.


================= Trying to gnaw off my leg to get out of this trap ========================

Hegelian dialectic:  The evidence that I am running on is that my Ubuntu (currently 20.10, but
for several versions past), the current Repository test passes both Standard & Daylight.

I can post some logs of some simple C programs. 

You mentioned that you are seeing the "local timezone" behavior on two
separate operating systems.  Can you say (or private message) which
ones (and versions)?

Reply:
Not to be annoying, just can not stop gnawing at the bone...

Let me send up a trial balloon. The wording can be improves, I 
am going for concept.    

While it would be satisfying (and helpful to future developers)
to identify root cause, perhaps we should follow the example
of Isaac Newton and feign no hypotheses.  That is, describe
behavior and not root cause.

We know, I believe, and it is important to convey:

1) On all know systems where the operating system time is
     "standard" (excluding Irish Standard time). the localtime
    tests perform as expected.

2) The Continuous Integration system runs on UTC. That
    is always "standard" time, so the localtime() tests execute
    in at least one place.

3) Differences in behavior of localtime*() have been observed
    when the Operating System time is not "standard".  These
    differences are unlikely to be due to Scala Native.

    The cause of these differences has not been tracked down. <<< Feel free to delete. 
                                                                                                  <<< Consensus  by omission  

   Due to these differences, the localtime*() tests are not executed
   when the Operating System time is not "standard".

Reply:
> Hegelian dialectic: The evidence that I am running on is that my Ubuntu (currently 20.10, but
for several versions past), the current Repository test passes both Standard & Daylight.

If it's not much indiscretion, what timezone are you in at?

> You mentioned that you are seeing the "local timezone" behavior on two
separate operating systems

OSX 10.15.7 and Arch Linux (rolling release)

Reply:
@ncordon Ah, the OS info helps.   Perhaps the comment can say something like (needs wordsmithing)
Ubuntu  appears to use the dst rules for the timestamp; OSX 10.15.7 & Arch Linux appear to 
use dst rules from the OS.   That is, say what we know.

My concern is that by saying "all" do it one way (OSX, Arch Linus), when there is a demonstrable
 counter-example (Ubuntu) will lead people to doubt, as they should, the credibility of the test.

As a thought example, we could pick up the OS name, now that we know them, and branch
on that (with a default of "only in standard"). That strikes me as over-engineering for a Test,
but it is good to be able to see far enough down the road to understand "what if".

The itch in my mind about "Why do two people report two different things" is now resolved,
thank you.

What are your thoughts about moving forward and putting this to bed?

Reply:
I added another line describing we observed differences in the `tm_isdst` between macOS, Arch and Ubuntu. I hope that covers it 🤞 . Goes without saying, but thanks a lot for the guidance :)

Reply:
@ncordon There were some "bump in the night" build problems over the weekend, since resolved
thanks to @ekrich.  Could you rebase this, when you can.  I believe that will give a clean
build.

I will close my WIP PR in favor of this one.  It is hammered copper and much better than
my mud & daub efforts.  I think that the FreeBSD changes will merge without bother,
but we can fix that if necessary.

re: "Goes without saying, but thanks a lot for the guidance :)"  Thank you for saying it anyway,
     probably made my day, if not my month.